apps/staking/app/stake/[address]/NewStake.tsx

```diff
@@ -93,14 +101,15 @@ export function NewStake({ contract }: { contract: ContributorContractInfo }) {
       }),
     },
     rewardsAddress: {
-      required: false,
+      required: !!vestingContract,
+      bannedAddresses: bannedRewardsAddresses,
     },
   });
 
   const form = useForm<StakeFormSchema>({
     resolver: zodResolver(formSchema),
     defaultValues: {
-      rewardsAddress: '',
+      rewardsAddress: vestingContract ? (connectedAddress ?? '') : '',
```

Comment:
```suggestion
      rewardsAddress: vestingContract || connectedAddress || '',
```

Reply:
This is not the same thing, we want:
- If actor is a vesting contract, use the connectedAddress if there is one else empty string
- If not connected, we want an empty string

Your suggestion changes the logic to use vesting contract address if it exists (which is banned, cant have it as a rewards address) :)